### DISCUSSION TITLE
Vercel: 405 error on Post Request

### DISCUSSION THREAD

**Discussion by sellmon** [vercel/community#3176] [Help]
### Summary

Hello, I don't understand what the error is.
Everything works fine in the local environment, but when I deployed it on Vercel, I encountered this error saying that the method is not allowed.

My stack: Monorepo, Next.js, TypeScript, Supabase.

___

```
...
try {
      console.log("Going to api/verify");
      console.log("POST request:", address);

      const response = await fetch("/api/verify", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          address
        }),
      });

      console.log("Data sended to api/verify");

      // I got error on this stage 👇 
      if (response.ok) {
        const data = await response.json();
        console.log("Response:", data.address);
      } else {
        throw new Error("Failed to verify address...");
      }
    } catch (error) {
      console.error(error);
    }
...
```

My api/verify/index.ts

```
//pages/apy/verify
import verifyTask from "../../../scripts/verifyTask";

export default function handler(req, res) {
  console.log("We are on api/verify");

  if (req.method === "POST") {
    const { address} = req.body;

    console.log("New address: ", address);

    // Call verification
    verifyTask(address);

    res.status(200).json({
      message: "Server response: 200",
      methods: ["POST"],
      address: address,
      taskCompleted: true,
    });
  } else {
    console.log("Error while sending POST-request");
    res.status(405).json({ message: "Method Not Allowed" });
  }
}

```

What I'm doing wrong? 

I see some similar problems, but didn't see solution
https://github.com/orgs/vercel/discussions?discussions_q=405

Do I need have a vercel.json in my monorepo to represent all my endpoints?
https://github.com/orgs/vercel/discussions/2470 - but here in didn't help

I stuck here.

### Steps to Reproduce

You can find various issues related to the 405 error topic through the link provided above, but for example, here:
https://github.com/orgs/vercel/discussions/2632

There is an issue with axios, but it doesn't exist with fetch. However, in my case, I am using fetch. In short, I don't understand where to look for a solution. Can someone explain where to dig deeper?

**Comment by amyegan** [comment]
Hi @sellmon. Can you share a public [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I want to be sure I'm using the same code and packages that you are, and the MRE will ensure we are looking at the same thing 🙏

**Reply by amyegan** [reply]
From the code already provided, it looks like you have an if/else block that returns `405: Method Not Allowed` if `req.method` is not `POST`. If you haven't already tried it, you could add some extra logging there to see if the request is coming through with an unexpected method.

**Reply by sellmon** [reply]
Hi @amyegan , the problem turned out to be deeper. As you can see, the verifyTask function is being called on the API, and when it runs, it returns a 405 error.

Within the function, there is a child_process that goes to a neighboring repository to run a script for contract verification.

My monorepo structure is as follows:

```
    apps
        contracts
        web
    packages
```
And here lies the error that I don't know how to solve. I am in the "web" directory and trying to execute child_process in the "contracts" directory because that's where all the contract-related environment is.
```
Uncaught Exception {"errorType":"Error","errorMessage":"spawn npx ENOENT","code":"ENOENT","errno":-2,"syscall":"spawn npx","path":"npx","spawnargs":["hardhat","verify","0x2bFf3fD269Fc7C1139906Ea2dbfbE47DBc8f2861","0x4C6f078E9F03910Fd1d3E2B2524A2Ee1f4fB4234","534768073287171",86400,0],"stack":["Error: spawn npx ENOENT","at ChildProcess._handle.onexit (node:internal/child_process:283:19)","at onErrorNT (node:internal/child_process:476:16)","at process.processTicksAndRejections (node:internal/process/task_queues:82:21)"]}
Unknown application error occurred
Runtime.Unknown
```
This error is related to the fact that there is no environment for child_process.

In general, I'm stuck. Now I understand that I need to configure vercel.json, or set up tasks for TurboRepo, or somehow install dependencies in the "contracts" directory so that the script can run.

I have tried:

     - turbo build && cd ../contracts && pnpm install (but it gives an error)
     - running a new project separately with "contracts," but then it becomes a separate application and requires installing "next."
     - currently trying to create the correct configuration.

If you have any suggestions on how to proceed, it would be great.

Also, you can probably close this topic as the error is deeper, and I have created a new ticket. If you could pay attention to it, I would be grateful because I'm currently dealing with this particular issue: [link to the GitHub ticket](https://github.com/vercel/turbo/discussions/5495).

**Reply by rileycleavenger** [reply]
Did you ever figure this out?

**Comment by Sushant0999** [comment]
I have also the same issue but after searching I got this solution, I haven't tried but I think this will work.
Issue : https://github.com/orgs/vercel/discussions/3911
